core-rust/state-manager/src/state_computer.rs

```diff
@@ -1210,12 +1271,55 @@ where
         self.committed_transactions_metrics
             .update(transactions_metrics_data);
 
+        let mut locked_protocol_state = self.protocol_state.write();
+        *locked_protocol_state = new_protocol_state;
+        drop(locked_protocol_state);
+
         Ok(CommitSummary {
             validator_round_counters: round_counters,
             num_user_transactions,
         })
     }
 
+    pub fn apply_protocol_update(
+        &self,
+        protocol_updater: &dyn ProtocolUpdater,
+    ) -> ProtocolUpdateResult {
+        let mut locked_ledger_transaction_validator = self.ledger_transaction_validator.write();
+        *locked_ledger_transaction_validator = protocol_updater
+            .state_computer_configurator()
+            .ledger_transaction_validator();
+        drop(locked_ledger_transaction_validator);
```

Comment:
Using scoping rather than `drop` for all of these:
```
  {
        let mut locked_ledger_transaction_validator = self.ledger_transaction_validator.write();
        *locked_ledger_transaction_validator = protocol_updater
            .state_computer_configurator()
            .ledger_transaction_validator();
  }
```